backend/src/apiserver/server/job_server_test.go

```diff
@@ -162,7 +157,60 @@ func TestValidateApiJob_NoValidPipelineSpecOrPipelineVersion(t *testing.T) {
 	}
 	err := server.validateCreateJobRequest(&api.CreateJobRequest{Job: apiJob})
 	assert.Equal(t, codes.InvalidArgument, err.(*util.UserError).ExternalStatusCode())
-	assert.Contains(t, err.Error(), "Neither pipeline spec nor pipeline version is valid")
+	assert.NotNil(t, err)
+	assert.Contains(t, err.Error(), "Please specify a pipeline by providing a pipeline spec or/and pipeline version.")
+}
+
+func TestValidateApiJob_WorkflowManifestAndPipelineVersion(t *testing.T) {
```

Comment:
Not sure whether I need to duplicate the UT for job and run since the UT of such logic have been covered in UT of validatePipelineSpecAndCheckPipelineVersionReference

Reply:
discussed offline. Cases got covered in UT of validatePipelineSpecAndResourceReferences. Test some cases double in UT of job and run.